pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2496,6 +2525,34 @@ protected void handleEndTxn(CommandEndTxn command) {
                 });
     }
 
+    private CompletableFuture<Boolean> verifyTxnOwnershipForTCToBrokerCommands() {
+        if (service.isAuthenticationEnabled() && service.isAuthorizationEnabled()) {
+            return getBrokerService()
+                    .getAuthorizationService()
+                    .isSuperUser(getPrincipal(), getAuthenticationData());
```

Comment:
Does this work correctly for with proxy and original auth data? I will try to review more closely tomorrow.

Reply:
One issue with this code is that we should always get this data from the netty event loop that the channel runs on.